### ISSUE TITLE
Latest update fails to apply

### ISSUE THREAD

**Issue by javahaxxor** [rockstor/rockstor-core#634]
From rockstor.log

Traceback (most recent call last):
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/core/handlers/base.py", line 114, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/views/decorators/csrf.py", line 57, in wrapped_view
    return view_func(*args, **kwargs)
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/views/generic/base.py", line 69, in view
    return self.dispatch(request, *args, **kwargs)
  File "/opt/rockstor/eggs/djangorestframework-2.4.3-py2.7.egg/rest_framework/views.py", line 403, in dispatch
    response = self.handle_exception(exc)
  File "/opt/rockstor/eggs/djangorestframework-2.4.3-py2.7.egg/rest_framework/views.py", line 391, in dispatch
    self.initial(request, *args, **kwargs)
  File "/opt/rockstor/eggs/djangorestframework-2.4.3-py2.7.egg/rest_framework/views.py", line 320, in initial
    self.perform_authentication(request)
  File "/opt/rockstor/eggs/djangorestframework-2.4.3-py2.7.egg/rest_framework/views.py", line 270, in perform_authentication
    request.user
  File "/opt/rockstor/eggs/djangorestframework-2.4.3-py2.7.egg/rest_framework/request.py", line 226, in user
    self._authenticate()
  File "/opt/rockstor/eggs/djangorestframework-2.4.3-py2.7.egg/rest_framework/request.py", line 401, in _authenticate
    user_auth_tuple = authenticator.authenticate(self)
  File "/opt/rockstor/eggs/djangorestframework-2.4.3-py2.7.egg/rest_framework/authentication.py", line 117, in authenticate
    if not user or not user.is_active:
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/utils/functional.py", line 213, in inner
    self._setup()
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/utils/functional.py", line 298, in _setup
    self._wrapped = self._setupfunc()
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/contrib/auth/middleware.py", line 18, in <lambda>
    request.user = SimpleLazyObject(lambda: get_user(request))
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/contrib/auth/middleware.py", line 10, in get_user
    request._cached_user = auth.get_user(request)
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/contrib/auth/__init__.py", line 140, in get_user
    user_id = request.session[SESSION_KEY]
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/contrib/sessions/backends/base.py", line 47, in __getitem__
    return self._session[key]
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/contrib/sessions/backends/base.py", line 173, in _get_session
    self._session_cache = self.load()
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/contrib/sessions/backends/db.py", line 20, in load
    expire_date__gt=timezone.now()
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/db/models/manager.py", line 151, in get
    return self.get_queryset().get(*args, **kwargs)
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/db/models/query.py", line 301, in get
    num = len(clone)
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/db/models/query.py", line 77, in __len__
    self._fetch_all()
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/db/models/query.py", line 854, in _fetch_all
    self._result_cache = list(self.iterator())
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/db/models/query.py", line 220, in iterator
    for row in compiler.results_iter():
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/db/models/sql/compiler.py", line 709, in results_iter
    for rows in self.execute_sql(MULTI):
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/db/models/sql/compiler.py", line 781, in execute_sql
    cursor = self.connection.cursor()
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/db/backends/__init__.py", line 159, in cursor
    cursor = util.CursorWrapper(self._cursor(), self)
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/db/backends/__init__.py", line 129, in _cursor
    self.ensure_connection()
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/db/backends/__init__.py", line 124, in ensure_connection
    self.connect()
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/db/utils.py", line 99, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/db/backends/__init__.py", line 124, in ensure_connection
    self.connect()
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/db/backends/__init__.py", line 112, in connect
    self.connection = self.get_new_connection(conn_params)
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/db/backends/postgresql_psycopg2/base.py", line 116, in get_new_connection
    return Database.connect(**conn_params)
  File "/opt/rockstor/eggs/psycopg2-2.5.2-py2.7-linux-x86_64.egg/psycopg2/__init__.py", line 164, in connect
    conn = _connect(dsn, connection_factory=connection_factory, async=async)
OperationalError: FATAL:  the database system is starting up

**Comment by freeurmind** [comment]
If you do a #yum update, will you see following:

Error: Package: 4:netatalk-3.1.6-0.0.1.el2.x86_64 (@rockstor)
           Requires: /bin/perl
           Removing: 4:perl-5.16.3-283.el7.x86_64 (@anaconda/3)
               Not found
           Updated By: 4:perl-5.16.3-285.el7.x86_64 (base)
               Not found
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest

**Comment by javahaxxor** [comment]
That solved it.  So if the Web update features depends on yum,  it should also show the error messages from yum,  which it doesn't

**Comment by schakrava** [comment]
@javahaxxor good point. Currently, update is one synchronous call and the web request from the ui usually times out before the actual update finishes making it hard to show the yum error. We've recently added async job handler for long running tasks like this. We'll refactor yum operations to be async. Then, it will be a bit easier to capture and display the error appropriately no matter how long it takes to finish or error out. Thanks for reporting this issue.

I also see the DB system is starting up... problem as a separate issue. I am curious, what are your machine specs? Another user running rockstor on a vm had this problem.

**Comment by schakrava** [comment]
@freeurmind Thanks for the workaround. I just updated on one of my test vms here and could not reproduce the problem. In my case, the Perl package was updated from 283 to 285 without any glitches as part of the update. What do you think the reason is behind you and @javahaxxor running into the yum update error? Is it a client/network error? I'd like to know if it's reproducible.

**Comment by javahaxxor** [comment]
@schakrava it's a HP microserver, the base model

**Comment by schakrava** [comment]
Thanks everyone. Closing this issue as all problems mentioned here are not valid anymore.
